Agenda/Controllers/Agenda/GoalDetailsViewController.swift

```diff
@@ -0,0 +1,119 @@
+        
+        setupViewAndConstraints()
+    }
+    
+    @objc private func closeButtonTapped() {
+        dismiss(animated: true)
+    }
+    
+    @objc private func saveButtonTapped() {
+        coreDataManager.rewriteGoal(data: goalData, in: goal)
+        delegate?.reloadTableView()
+        
+        // TODO: Fix not updating HistoryTableView on ending up the goal (current >= aim)
+        // TODO: Display some kind of SPAlert https://t.me/sparrowcode/120
+    }
+    
+    private func setupViewAndConstraints() {
+        view.addSubview(tableView)
+        
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+    }
+}
+
+// MARK: - UITableViewDelegate, UITableViewDataSource
+extension GoalDetailsViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        2
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        2
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: GoalTableViewCell.identifier, for: indexPath) as? GoalTableViewCell else {
+            fatalError("Error")
+        }
+        cell.goal = goalData
+        cell.configure(indexPath: indexPath)
+        cell.delegate = self
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        UITableView.automaticDimension
+    }
+}
+
+extension GoalDetailsViewController: GoalTableViewCellDelegate {
+    
+    func checkBarButtonEnabled() {
+        // TODO: Check for 'Save' button is enabled (if any changes were done)
+    }
+    
+    // Update height of UITextView based on string height
+    func updateHeightOfRow(_ cell: GoalTableViewCell, _ textView: UITextView) {
+        let size = textView.bounds.size
+        let newSize = tableView.sizeThatFits(CGSize(width: size.width, height: CGFloat.greatestFiniteMagnitude))
+        
+        if size.height != newSize.height {
+            UIView.setAnimationsEnabled(false)
+            tableView.beginUpdates()
+            tableView.endUpdates()
+            UIView.setAnimationsEnabled(true)
+            // Scoll up your textview if required
+            if let thisIndexPath = tableView.indexPath(for: cell) {
+                tableView.scrollToRow(at: thisIndexPath, at: .bottom, animated: false)
+            }
+        }
+    }
```

Comment:
This method has already been declared. It's better to refactor it to use only one

Reply:
Resolved one-two days ago